soc/tsi/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 TSI
+# SPDX-License-Identifier: Apache-2.0
+
+if(DEFINED SOC_SERIES)
+  add_subdirectory(${SOC_SERIES})
+else()
+  add_subdirectory(${SOC_NAME})
+endif()
```

Comment:
There are few changes I have made in top level CMakeLists..txt, see if some of that can be moved here is that change belongs here for our own requirements for creating FPGA hex file. Maybe then we do not need config flags.

Reply:
I didn't change the top-level CMakeLists.txt file. This file is different from the Zephyr CMakeLists.txt.